mlx/io/load.cpp

```diff
@@ -226,10 +227,7 @@ array load(std::shared_ptr<io::Reader> in_stream, StreamOrDevice s) {
     throw std::runtime_error("[load] Failed to open " + in_stream->label());
   }
 
-  auto stream = to_stream(s, Device::cpu);
-  if (stream.device != Device::cpu) {
-    throw std::runtime_error("[load] Must run on a CPU stream.");
-  }
+  auto stream = to_stream(s, cu::is_available() ? Device::gpu : Device::cpu);
```

Comment:
`Device::gpu` is always device 0, it should load on the device that `s` is on.

Reply:
This is just the default if the stream is not specified. If the stream is specified it will use it's device.